libs/utils/src/filter.c

```diff
@@ -165,7 +165,9 @@ static celix_filter_t * filter_parseNot(char * filterString, int * pos) {
     }
 
     child = filter_parseFilter(filterString, pos);
-
+    if(child == NULL){
```

Comment:
styling nitpick: missing some spaces.
```suggestion
    if (child == NULL) {
```

Reply:
I am planning to create a support develop branch for a 2.4.1 release next week and will also add the changes of this PR (cherry pick)